ui/ui-components/components/record/AddSampleRecordForm.tsx

```diff
@@ -0,0 +1,77 @@
+import { useMemo, useState } from "react";
+import { Form } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { useGrouparooModelContext } from "../../contexts/grouparooModel";
+import { ApiHook } from "../../hooks/useApi";
+import { Actions, Models } from "../../utils/apiData";
+import LoadingButton from "../LoadingButton";
+
+interface Props {
+  properties: Models.PropertyType[];
+  onSubmitComplete: (record?: any) => void;
+  execApi: ApiHook["execApi"];
+}
+
+const AddSampleRecordForm: React.FC<Props> = ({
+  execApi,
+  onSubmitComplete,
+  properties,
+}) => {
+  const model = useGrouparooModelContext();
+  const { handleSubmit, register } = useForm();
+  const [submitting, setSubmitting] = useState(false);
+
+  const propertiesWithPrimaryKey = useMemo(
+    () =>
+      properties.filter((property) => property.unique && property.isPrimaryKey),
+    [properties]
+  );
+
+  const onSubmit: Parameters<typeof handleSubmit>[0] = async (data) => {
+    setSubmitting(true);
+    const response = await execApi<Actions.RecordCreate>("post", `/record`, {
+      modelId: model.id,
+      properties: { [data.uniqueProperty]: data.value },
+    });
+    setSubmitting(false);
+    onSubmitComplete(response?.record);
+  };
+
+  return (
+    <Form id="form" onSubmit={handleSubmit(onSubmit)}>
+      <Form.Group>
+        <Form.Label>Unique Property</Form.Label>
+        <Form.Control
+          name="uniqueProperty"
+          as="select"
+          ref={register}
+          disabled={submitting}
+        >
+          {propertiesWithPrimaryKey.map((rule) => (
+            <option key={`rule-${rule.key}`}>{rule.key}</option>
+          ))}
+        </Form.Control>
+      </Form.Group>
+      <Form.Group>
+        <Form.Label>Value</Form.Label>
+        <Form.Control
+          autoFocus
+          required
+          type="text"
```

Comment:
I need to improve the validation for this because I noticed that I can create really bad data. I could probably take a look at the property type itself and determine what type of validation should happen.